python/pyspark/pandas/generic.py

```diff
@@ -978,6 +981,9 @@ def to_json(
         0         a
         1         c
         """
+        default_options: Dict[str, Any] = {"ignoreNullFields": False}
+        options = {**default_options, **options}
```

Comment:
Merging the default_options should be after line 988.

```py
        if "options" in options and isinstance(options.get("options"), dict) and len(options) == 1:
            options = options.get("options")
        options = {**default_options, **options}
```

otherwise the if-clause never takes effect.

The if-clause is to help misuse of `options` keyword argument, e.g.,:

```py
psdf.to_json( ..., options={"ignoreNullFields": True})
```